sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -700,6 +700,8 @@ object ScalaReflection extends ScalaReflection {
         Schema(TimestampType, nullable = true)
       case t if isSubtype(t, localTypeOf[java.time.LocalDate]) => Schema(DateType, nullable = true)
       case t if isSubtype(t, localTypeOf[java.sql.Date]) => Schema(DateType, nullable = true)
+      case t if isSubtype(t, localTypeOf[CalendarInterval]) =>
+        Schema(CalendarIntervalType, nullable = true)
```

Comment:
Hey,

> Since 3.0.0, we make CalendarInterval public ...

Can we clarify which scope we want to change? According to your PR description, we should support read and write paths too, which I think we decided to don't implement.

Reply:
Oh, I separate this to the 3rd PR question, I will duplicate it in the first one.